### PR TITLE
New version: Amazon.WorkspacesClient version 5.2.1

### LINKED ISSUE #70727: [Package Issue]: Amazon.WorkSpacesClient
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installer hash mismatch.

### Brief description of your issue

There is a hash mismatch between the manifest for Amazon WorkSpaces Client and the latest MSI that is available to download.

The manifest needs to be updated to reflect the latest version that is now available to download.

Current version in manifest: 5.2.0.3089
Current latest version: 5.2.1.3123

We are unable to use --force as we run as admin

### Steps to reproduce

winget install --exact --id Amazon.WorkspacesClient --silent --accept-package-agreements --accept-source-agreements --scope Machine

### Actual behavior

Hash mismatch, winget will not install application

### Expected behavior

Hash should match and application should install

### Environment

```shell
Windows Package Manager v1.3.2091
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19044.1889
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2091.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by Trenly**
﻿- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

* Resolves #70727

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/70765)

### CODE REVIEW

File: manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-PackageIdentifier: Amazon.WorkspacesClient
-PackageVersion: 5.2.0.3089
-DefaultLocale: en-US
-ManifestType: version
-ManifestVersion: 1.1.0
+# Created with YamlCreate.ps1 v2.1.3 $debug=AUSU.5-1-19041-1682
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.2.0.schema.json
+
+PackageIdentifier: Amazon.WorkspacesClient
+PackageVersion: 5.2.1
```
Comment (JamieSabbatella) [inline manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.yaml:5]: Version should be: 5.2.1.3123

File: manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.locale.en-US.yaml [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
-- amazon
-- rdp
-- workspaces
-ManifestType: defaultLocale
-ManifestVersion: 1.1.0
+# Created with YamlCreate.ps1 v2.1.3 $debug=AUSU.5-1-19041-1682
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: Amazon.WorkspacesClient
+PackageVersion: 5.2.1
```
Comment (JamieSabbatella) [inline manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.locale.en-US.yaml:5]: Version should be: 5.2.1.3123

File: manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.installer.yaml [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
-  InstallerUrl: https://d2td7dqidlhjx7.cloudfront.net/prod/global/windows/Amazon+WorkSpaces.msi
-  InstallerSha256: ACFABF83EA29929D58289CEF9FC49000C127F5245A0E80C6B85E2DB9B44A5298
-  ProductCode: '{BFF45D60-ED93-489D-AB67-B8720715D10B}'
-ManifestType: installer
-ManifestVersion: 1.1.0
+# Created with YamlCreate.ps1 v2.1.3 $debug=AUSU.5-1-19041-1682
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: Amazon.WorkspacesClient
+PackageVersion: 5.2.1
```
Comment (JamieSabbatella) [inline manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.installer.yaml:5]: Version should be: 5.2.1.3123

**Review by JamieSabbatella — CHANGES_REQUESTED** [review]
New version should be: 5.2.1.3123

File: manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.yaml (hunk 1 above)
Comment (Trenly) [reply manifests/a/Amazon/WorkspacesClient/5.2.1/Amazon.WorkspacesClient.yaml:5]: No. The marketed version is 5.2.1, the DisplayVersion is 5.2.1.3123, which is shown in the AppsAndFeaturesEntries

**Comment by Trenly** [comment]
> New version should be: 5.2.1.3123

The marketed version is 5.2.1, the DisplayVersion is 5.2.1.3123, which is shown in the AppsAndFeaturesEntries. If you view the release notes on their website, they always refer to only the first three digits. Therefore, 5.2.1 is correct in this specific case

**Comment by JamieSabbatella** [comment]
> > New version should be: 5.2.1.3123
> 
> The marketed version is 5.2.1, the DisplayVersion is 5.2.1.3123, which is shown in the AppsAndFeaturesEntries. If you view the release notes on their website, they always refer to only the first three digits. Therefore, 5.2.1 is correct in this specific case



> > New version should be: 5.2.1.3123
> 
> The marketed version is 5.2.1, the DisplayVersion is 5.2.1.3123, which is shown in the AppsAndFeaturesEntries. If you view the release notes on their website, they always refer to only the first three digits. Therefore, 5.2.1 is correct in this specific case

image

I disagree as all previous versions have been a full version, as in the picture above. This could potentially lead to issues if a new build is released.

**Review by ItzLevvie — APPROVED** [review]

**Comment by Trenly** [comment]
> I disagree as all previous versions have been a full version, as in the picture above. This could potentially lead to issues if a new build is released.

When they release a new build, they remove the old build, which is why there is only one version in the previous version list. The reason the full version was always used before was because there was not the ability for the client to use the AppsAndFeaturesEntries to determine the version. However, the 1.3 client does have that ability. 

The version you showed in your screenshot is actually referred to by AWS as `5.2.0` in their documentation - https://docs.aws.amazon.com/workspaces/latest/userguide/amazon-workspaces-windows-client.html#windows-release-notes

@officialesco @jedieaston @ItzLevvie - Any thoughts or feedback here?

**Comment by JamieSabbatella** [comment]
> 
> When they release a new build, they remove the old build, which is why there is only one version in the previous version list. The reason the full version was always used before was because there was not the ability for the client to use the AppsAndFeaturesEntries to determine the version. However, the 1.3 client does have that ability.
> 
> The version you showed in your screenshot is actually referred to by AWS as `5.2.0` in their documentation - https://docs.aws.amazon.com/workspaces/latest/userguide/amazon-workspaces-windows-client.html#windows-release-notes
> 
> @OfficialEsco @jedieaston @ItzLevvie - Any thoughts or feedback here?

Yes, I understand what you are saying, but I believe using a full version of the files & app would be the correct course of action and appears to be standard across all of the MS apps, and all Amazon CLI and previous versions on this client

**Comment by jedieaston** [comment]
I agree with @Trenly, when you ask Amazon for support they’re going to ask for the version number in the release notes, not the ARP version. Previous versions should be able to upgrade fine since winget will rematch them.

**Pull request merged**